src/Handler/StreamHandler.php

```diff
@@ -286,7 +288,7 @@ private function createStream(RequestInterface $request, array $options)
             foreach ($options as $key => $value) {
                 $method = "add_{$key}";
                 if (isset($methods[$method])) {
-                    $this->{$method}($request, $context, $value, $params);
+                    $this->{$method}($request, $context, $value, $params, $uri);
```

Comment:
Ah, okay, you are totally right. For some reasons I thought you are trying to set the host in the original request. Even then, adding the URI to all methods bellow seems to be too much. How about a `resolveUri` or `resolveAddress` or something similar method? It would make things easier IMO.